code/filealerter/pcapdecoder.q

```diff
@@ -0,0 +1,98 @@
+// Decoder is designed for pcap version 2.4
+// Info on pcap structure => https://www.kroosec.com/2012/10/a-look-at-pcap-file-format.html
+
+\d .pcap
+
+// size of headers in bytes and dict of protocol code conversions
+globheader: 24;
+packetheader: 16;
+
+// mapping from protocol id numbers to protocol name
+// add desired protocols to dictionary for decoder to be able to translate their id's
+allcodes:(enlist 6)!(enlist `TCP);
+
+
+// returns table of packet data
+buildtable:{[file]
+ // check that version number of input file is 2.4
+ pcapversioncheck: all 2 0 4 0 = "h"$4#4_read1 file;
+
+ // if version is correct, gettablerow is iterated over each datapacket, extracting data
+ $[pcapversioncheck;[
+  data:1_ last each {[filebinary] // initial x and binary starting numbers removed from array list to make table
+   filebytesize: count filebinary;
+   // x here is a list containing starting binary point for packet (x[0]) and row of data for that packet (x[1])
+   gettablerow[filebinary;]\[{y>(first x[0])+40}[;filebytesize];(),0]    
+   } read1 file;
+ 
+  data: update sym:` from data;
+  `time`sym xcols data];
+  .lg.o[`alerter;"pcap version number of ", file, " is incorrect, so could not be decoded"]
+  ]
+ }
+
+
+// returns starting point of next packet and row data
+gettablerow:{[filebinary;x]  // data for a single row
+ // x is a list containing starting binary point for packet (x[0]) and row of data for that packet (x[1])
+ time:     gettime[filebinary;x];
+ flags:    getflags[filebinary;x];
+ protocol: getprotocol[filebinary;x];
+ info:     getinfo[filebinary;x];
+ ips:      getips[filebinary;x];
+
+ totallength: 0x0 sv datafromfile[filebinary;x;18;2];
+ IPheader:    4*"J"$last string filebinary[x[0]+globheader+packetheader+16];
+ TCPheader:   4* first "0123456789abcdef"?/:string filebinary[x[0]+globheader+packetheader+48];
+ len:         (first first totallength - IPheader + TCPheader) mod 65536;
+
+ length: first raze ((enlist "h";enlist 2)1: filebinary[x[0]+36 37]) mod 65536;
```

Comment:
I think this one could be `0x0 sv` as well?

Reply:
i dont think this one works the same way since the binary here is little endian so it is read using `(enlist "h";enlist 2)1:` rather than the previous big endian examples: `(enlist 2;enlist"h")1:`. Ill have a look at possible simplifications though

Reply:
Ah, I missed the endianness, fair enough

Reply:
sv seems to work with little endian binary as long as you reverse the binary first: 
`(0x0 sv reverse filebinary[x[0]+36 37]) mod 65536`